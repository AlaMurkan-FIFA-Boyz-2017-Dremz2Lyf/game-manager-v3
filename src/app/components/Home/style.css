.home {
  color: #000;
}

.sidebar-pusher {
  height: 100vh;
}

.sidebar {
  width: 100% !important;
}
