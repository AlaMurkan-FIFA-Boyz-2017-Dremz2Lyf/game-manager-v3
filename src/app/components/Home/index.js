import React, { Component } from 'react';
import { Segment, Header, Menu } from 'semantic-ui-react';
import Tournaments from '../Tournaments';
import Play from '../Play';

import './style.css';

class Home extends Component {
  state = {
    tournament: false,
  }

  onSelect = (e) => {
    const { tournament } = this.state;

    this.setState({
      tournament: !tournament,
    });
  }

  render() {
    const { tournament } = this.state;

    return (
      <div styleName="home">
        <Header textAlign="left" size="huge">Game Manager</Header>
        <Menu inverted vertical fixed="left">
          <Tournaments onSelect={this.onSelect} />
        </Menu>
        <Segment basic>
          <Play tournament={tournament} />
        </Segment>
      </div>
    );
  }
}

export default Home;
