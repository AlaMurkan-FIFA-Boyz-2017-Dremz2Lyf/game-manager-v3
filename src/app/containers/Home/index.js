import { gql, graphql, compose } from 'react-apollo';
import Home from '../../components/Home';

const getPlayer = gql`
query getPlayer($id: String!) {
  getPlayer(id: $id) {
    id
    username
  }
}
`;

export const getPlayerOptions = props => ({
  variables: {
    id: props.playerId,
  },
});

export const getPlayerProps = ({ ownProps, data: { loading, player } }) => ({
  ...ownProps,
  loading,
  player,
});

const HomeContainer = compose(
  graphql(getPlayer, {
    props: getPlayerProps,
    options: getPlayerOptions,
  }),
)(Home);

export default HomeContainer;
