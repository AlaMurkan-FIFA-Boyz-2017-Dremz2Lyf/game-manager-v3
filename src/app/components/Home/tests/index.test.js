import { shallow } from 'enzyme';
import React from 'react';

import Home from '../index';

describe('<Home />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });
  it('should match snapshot without a tournament', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with a tournament', () => {
    wrapper.setState({ tournament: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an onSelect method that changes state', () => {
    expect(wrapper.state('tournament')).toBe(false);
    wrapper.instance().onSelect();
    expect(wrapper.state('tournament')).toBe(true);
  });
});
