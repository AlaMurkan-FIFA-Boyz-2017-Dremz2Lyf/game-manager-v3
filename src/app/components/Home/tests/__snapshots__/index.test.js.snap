// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot with a tournament 1`] = `
<div
  className="style__home-2dfZi"
>
  <Segment>
    <Grid
      basic="true"
      horizontal="true"
    >
      <GridRow>
        <GridColumn
          width={2}
        >
          <Button
            as="button"
            icon="chevron left"
            onClick={[Function]}
          />
        </GridColumn>
        <GridColumn
          width={12}
        >
          <Header
            size="large"
            textAlign="center"
          >
            Game Manager
          </Header>
        </GridColumn>
        <GridColumn
          width={2}
        />
      </GridRow>
    </Grid>
  </Segment>
  <SidebarPushable
    as={[Function]}
    basic={true}
    className="style__sidebar-pusher-PBx9e"
  >
    <Sidebar
      animation="overlay"
      className="style__sidebar-1wYym"
      direction="right"
      visible={true}
    >
      <Play
        tournament={true}
      />
    </Sidebar>
    <SidebarPusher>
      <Tournaments
        onSelect={[Function]}
      />
    </SidebarPusher>
  </SidebarPushable>
</div>
`;

exports[`<Home /> should match snapshot without a tournament 1`] = `
<div
  className="style__home-2dfZi"
>
  <Segment>
    <Grid
      basic="true"
      horizontal="true"
    >
      <GridRow>
        <GridColumn
          width={2}
        />
        <GridColumn
          width={12}
        >
          <Header
            size="large"
            textAlign="center"
          >
            Game Manager
          </Header>
        </GridColumn>
        <GridColumn
          width={2}
        />
      </GridRow>
    </Grid>
  </Segment>
  <SidebarPushable
    as={[Function]}
    basic={true}
    className="style__sidebar-pusher-PBx9e"
  >
    <Sidebar
      animation="overlay"
      className="style__sidebar-1wYym"
      direction="right"
      visible={false}
    >
      <Play
        tournament={false}
      />
    </Sidebar>
    <SidebarPusher>
      <Tournaments
        onSelect={[Function]}
      />
    </SidebarPusher>
  </SidebarPushable>
</div>
`;
